bss/sequencer/driver.go

```diff
@@ -0,0 +1,205 @@
+package sequencer
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimistic-specs/bss/db"
+	"github.com/ethereum-optimism/optimistic-specs/l2os/rollupclient"
+	"github.com/ethereum-optimism/optimistic-specs/l2os/txmgr"
+	"github.com/ethereum-optimism/optimistic-specs/opnode/node"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type Config struct {
+	Name              string
+	L1Client          *ethclient.Client
+	L2Client          *ethclient.Client
+	RollupClient      *rollupclient.RollupClient
+	MinL1TxSize       uint64
+	MaxL1TxSize       uint64
+	BatchInboxAddress common.Address
+	HistoryDB         db.HistoryDatabase
+	ChainID           *big.Int
+	PrivKey           *ecdsa.PrivateKey
+}
+
+type Driver struct {
+	cfg        Config
+	walletAddr common.Address
+
+	currentBatch *node.BatchBundleResponse
+}
+
+func NewDriver(cfg Config) (*Driver, error) {
+	walletAddr := crypto.PubkeyToAddress(cfg.PrivKey.PublicKey)
+
+	return &Driver{
+		cfg:        cfg,
+		walletAddr: walletAddr,
+	}, nil
+}
+
+// Name is an identifier used to prefix logs for a particular service.
+func (d *Driver) Name() string {
+	return d.cfg.Name
+}
+
+// WalletAddr is the wallet address used to pay for transaction fees.
+func (d *Driver) WalletAddr() common.Address {
+	return d.walletAddr
+}
+
+// GetBlockRange returns the start and end L2 block heights that need to be
+// processed. Note that the end value is *exclusive*, therefore if the returned
+// values are identical nothing needs to be processed.
+func (d *Driver) GetBlockRange(
+	ctx context.Context,
+) (*big.Int, *big.Int, error) {
+
+	// Clear prior batch, if any.
+	d.currentBatch = nil
+
+	history, err := d.cfg.HistoryDB.LoadHistory()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ancestors := history.Ancestors()
+	log.Info("Ancestors", "ancestors", ancestors)
+	batchResp, err := d.cfg.RollupClient.GetBatchBundle(
+		ctx,
+		&node.BatchBundleRequest{
+			L2History: history.Ancestors(),
+			MinSize:   hexutil.Uint64(d.cfg.MinL1TxSize),
+			MaxSize:   hexutil.Uint64(d.cfg.MaxL1TxSize),
+		},
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Bundle is not available yet, return the tip of our known chain.
```

Comment:
The comment is confusing. It returns the `tip+1`, not the tip. Is that `+1` correct?